frontend/packages/ceph-storage-plugin/src/types.ts

```diff
@@ -33,24 +33,29 @@ export type StoragePoolKind = K8sResourceCommon & {
   };
 };
 
-export type OCSStorageClusterKind = K8sResourceCommon & {
+export type StorageClusterKind = K8sResourceCommon & {
   spec: {
-    monDataDirHostPath?: string;
     manageNodes: boolean;
-    storageDeviceSets: StorageDeviceSet[];
+    storageDeviceSets: DeviceSet[];
+    resources: StorageClusterResource;
+    encryption: {
+      enable: boolean;
+    };
+    monDataDirHostPath?: string;
   };
   status?: {
     phase: string;
   };
 };
 
-type StorageDeviceSet = {
+export type DeviceSet = {
   name: string;
   count: number;
   replica: number;
-  resources: {};
-  placement: {};
+  resources: ResourceConstraints;
+  placement?: any;
   portable: boolean;
+  encryption?: { [key: string]: any };
```

Comment:
```suggestion
  encryption?: { enable: boolean };
```
Shouldn't encryption be only part of spec?

Reply:
Right! It should not be here.